app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -17,37 +17,79 @@ import androidx.compose.foundation.lazy.GridCells
 import androidx.compose.foundation.lazy.LazyVerticalGrid
 import androidx.compose.foundation.lazy.items
 import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Snackbar
+import androidx.compose.material.SnackbarDuration
+import androidx.compose.material.SnackbarHost
+import androidx.compose.material.SnackbarResult
 import androidx.compose.material.Surface
+import androidx.compose.material.rememberScaffoldState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.launch
 import org.mozilla.fenix.R
 import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.wallpapers.Wallpaper
 
 /**
- * The screen for controlling settings around Wallpapers.
+ * The screen for controlling settings around Wallpapers. When a new wallpaper is selected,
+ * a snackbar will be displayed.
  *
  * @param wallpapers Wallpapers to add to grid.
  * @param selectedWallpaper The currently selected wallpaper.
- * @param onSelectWallpaper Action to take when a new wallpaper is selected.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ * @param onViewWallpaper Callback for when the view action is clicked from snackbar.
  */
 @Composable
 fun WallpaperSettings(
     wallpapers: List<Wallpaper>,
     selectedWallpaper: Wallpaper,
     onSelectWallpaper: (Wallpaper) -> Unit,
+    onViewWallpaper: () -> Unit,
 ) {
-    Surface(color = FirefoxTheme.colors.layer2) {
-        WallpaperThumbnails(
-            wallpapers = wallpapers,
-            selectedWallpaper = selectedWallpaper,
-            onSelectWallpaper = onSelectWallpaper
-        )
+    val coroutineScope = rememberCoroutineScope()
+    val scaffoldState = rememberScaffoldState()
+    val snackbarMessage = stringResource(id = R.string.wallpaper_updated_snackbar_message)
+    val snackbarActionLabel = stringResource(id = R.string.wallpaper_updated_snackbar_action)
```

Comment:
I would just inline these since the variable isn't used more than once.

Reply:
That was my thought as well, but `stringResource` can't be invoked outside of a `Composable` context, which the `launch`ed scope below is not. I think the alternative would be to capture the `LocalContext` and use it to resolve the strings, but that would lengthen the expression significantly.  We could add an extension function? It looks like there's already a `Context.getStringWithArgSafe` so there would be precedent to add a `Context.getString`

Reply:
Makes sense, I saw something similar in another PR. I wouldn't make the suggested changes in this PR. I think it's fine as-is for the purpose of this PR and we can revisit this in a followup if we feel strongly about it.

Reply:
This was resolved as a by-product of the font family changes you requested. I think we should consider creating a more generalized `FenixSnackbar` in a follow-up patch.

Reply:
Yep, I was thinking the same.

Reply:
I added #23221 to track that work.

Reply:
Cool, I was thinking of taking a stab at it if it's good with you.